core-tests/src/test/java/org/mule/MuleMessageDataTypePropagationTestCase.java

```diff
@@ -190,7 +190,8 @@ public void updatesTypeOnTransformation() throws Exception
 
         MuleEvent muleEvent = mock(MuleEvent.class);
 
-        transformationService.applyTransformers(muleMessage, muleEvent, Collections.singletonList(transformer));
+        muleMessage = (DefaultMuleMessage) transformationService.applyTransformers(muleMessage, muleEvent,
```

Comment:
No, because mimeType is set above and the setter is only available on MuleMessage.  I could define a second message instance for the transformation result though if you think that is best.